src/backend/access/common/reloptions.c

```diff
@@ -2001,10 +2003,16 @@ bytea *
 partitioned_table_reloptions(Datum reloptions, bool validate)
 {
 	if (validate && reloptions)
+	{
+		/* For babelfish partitioned table, allow usage of reloptions in TSQL dialect. */
+		if (pltsql_partitioned_table_reloptions_hook && pltsql_partitioned_table_reloptions_hook(reloptions))
```

Comment:
Should give a more reasonable name for the hook type like : 
```
pltsql_is_partitioned_table_reloptions_allowed_hook
```
the current name looks confusing.